fabric-compute-framework/src/main/java/com/olacabs/fabric/compute/processor/ProcessorBase.java

```diff
@@ -35,9 +35,10 @@
 public abstract class ProcessorBase {
     @Getter
     private final boolean isScheduled;
+
     @Getter
     @Setter
-    private int id;
+    private String id;
```

Comment:
This id is not used for acking. Actually it wasn't being used at all. Type of id here should be string because it should correspond to the id of the component set by user in the spec (human readable). There is one other id in PipelineStage which is internal id maintained by us for acking and that is an integer.